File: update_prices.py
import openpyxl
import asyncio
from pathlib import Path
import sys
sys.path.insert(0, '/home/ubuntu/icd10-search-engine')

async def update_prices():
    try:
        # Load Excel file
        wb = openpyxl.load_workbook('/home/ubuntu/upload/APPROVALFinalv15(3).xlsx')
        ws = wb.active
        
        # Import after path setup
        from server.db import getDb
        from drizzle.schema import drugLens
        from drizzle_orm import eq
        
        db = await getDb()
        updated = 0
        not_found = 0
        
        # Iterate through rows (skip header)
        for row_num in range(2, ws.max_row + 1):
            scientific_name = ws.cell(row_num, 1).value
            trade_name = ws.cell(row_num, 2).value
            price = ws.cell(row_num, 3).value
            
            if not trade_name or not price:
                continue
            
            # Try to find drug by trade name first
            result = await db.select().from_(drugLens).where(eq(drugLens.tradeName, trade_name)).limit(1)
            
            if not result and scientific_name:
                result = await db.select().from_(drugLens).where(eq(drugLens.scientificName, scientific_name)).limit(1)
            
            if result:
                # Update price
                await db.update(drugLens).set({'price': f'{price} SAR'}).where(eq(drugLens.id, result[0].id))
                updated += 1
                print(f"✓ Updated: {trade_name} - {price} SAR")
            else:
                not_found += 1
                print(f"✗ Not found: {trade_name}")
        
        print(f"\nSummary: {updated} updated, {not_found} not found")
    except Exception as e:
        print(f"Error: {str(e)}")

if __name__ == '__main__':
    asyncio.run(update_prices())

File: update_prices.ts
import * as XLSX from 'xlsx';
import { getDb } from './server/db';
import { drugLens } from './drizzle/schema';
import { eq } from 'drizzle-orm';

async function updatePrices() {
  try {
    // Load Excel file
    const file = XLSX.readFile('/home/ubuntu/upload/APPROVALFinalv15(3).xlsx');
    const ws = file.Sheets[file.SheetNames[0]];
    const data = XLSX.utils.sheet_to_json(ws);
    
    const db = await getDb();
    let updated = 0;
    let notFound = 0;
    
    // Iterate through rows
    for (const row of data) {
      const scientificName = row['SCIENTIFIC NAME'];
      const tradeName = row['TRADE NAME'];
      const price = row['PRICE (SAR)'];
      
      if (!tradeName || !price) continue;
      
      // Try to find drug by trade name first
      let result = await db
        .select()
        .from(drugLens)
        .where(eq(drugLens.tradeName, tradeName as string))
        .limit(1);
      
      if (!result.length && scientificName) {
        result = await db
          .select()
          .from(drugLens)
          .where(eq(drugLens.scientificName, scientificName as string))
          .limit(1);
      }
      
      if (result.length > 0) {
        // Update price - store as number only, no "SAR" suffix
        const priceValue = typeof price === 'number' ? price.toString() : price;
        await db
          .update(drugLens)
          .set({ price: priceValue })
          .where(eq(drugLens.id, result[0].id));
        updated++;
        console.log(`✓ Updated: ${tradeName} - ${priceValue}`);
      } else {
        notFound++;
        console.log(`✗ Not found: ${tradeName}`);
      }
    }
    
    console.log(`\nSummary: ${updated} updated, ${notFound} not found`);
    process.exit(0);
  } catch (error) {
    console.error('Error:', (error as Error).message);
    process.exit(1);
  }
}

updatePrices();
